manage_arkime/commands/add_vpc.py

```diff
@@ -7,19 +7,43 @@
 from manage_arkime.cdk_client import CdkClient
 import manage_arkime.constants as constants
 import manage_arkime.cdk_context as context
+from manage_arkime.vni_provider import SsmVniProvider, VniAlreadyUsed, VniOutsideRange, VniPoolExhausted
 
 logger = logging.getLogger(__name__)
 
-def cmd_add_vpc(profile: str, region: str, cluster_name: str, vpc_id: str, vni: int):
+def cmd_add_vpc(profile: str, region: str, cluster_name: str, vpc_id: str, user_vni: int):
     logger.debug(f"Invoking add-vpc with profile '{profile}' and region '{region}'")
 
     aws_provider = AwsClientProvider(aws_profile=profile, aws_region=region)
+    vni_provider = SsmVniProvider(cluster_name, aws_provider)
 
-    # Confirm the VNI is valid
-    if (vni <= constants.VNI_MIN) or (constants.VNI_MAX < vni):
-        logger.error(f"VNI {vni} is outside the acceptable range of {constants.VNI_MIN} to {constants.VNI_MAX} (inclusive)")
-        logger.warning("Aborting...")
-        return
+    # If the user didn't supply a VNI, try to find one
+    if not user_vni:
+        try:
+            next_vni = vni_provider.get_next_vni()
+        except VniPoolExhausted:
+            logger.error(f"There are no remaining VNIs in the range {constants.VNI_MIN} to {constants.VNI_MAX} to assign for this cluster")
```

Comment:
So is this just impossible to get to? Should it just wrap back to MIN? How would a user fix if they have this?

Reply:
It's possible to get to, just very unlikely.  If they reach this point, that means they'll have added 16,777,215 VPCs to a single Arkime Cluster.  If the need to add more than that, they can start a new cluster.  If they've deprovisioned some of those VPCs and want to re-use old VNIs, they can use the `--force-vni` option and manage the assignment themselves, and use the `list-vpcs` command to find the mapping between existing VPCs and their VNIs.  I previously had recycling of relinquished VNIs to make this marginally less likely to occur, but we decided to remove that feature to prevent pollution of data from old VPCs.

Reply:
Actually, I think it much more likely something else will break before a user hits this error message (e.g. scaling limits, AWS Account limits, etc), so practically speaking it's probably unreachable unless the user just constantly adds/removes VPCs.  Also, that process currently takes ~10 min/cycle, and, according to my calculator, that would means running it in a loop for 319 years.